src/test/store/transforms.test.js

```diff
@@ -0,0 +1,154 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import {
+  getProfileForUnfilteredCallTree,
+  getProfileForInvertedCallTree,
+} from '../fixtures/profiles/profiles-for-call-trees';
+import { storeWithProfile } from '../fixtures/stores';
+import {
+  addTransformToStack,
+  popTransformsFromStack,
+  changeInvertCallstack,
+} from '../../actions/profile-view';
+import { selectedThreadSelectors } from '../../reducers/profile-view';
+import { CallTree } from '../../profile-logic/call-tree';
+import type { IndexIntoCallNodeTable } from '../../types/profile-derived';
+
+export function formatTree(
+  callTree: CallTree,
+  children: IndexIntoCallNodeTable[] = callTree.getRoots(),
+  depth: number = 0,
+  previousString: string = ''
+) {
+  const whitespace = Array(depth * 2).join(' ');
+
+  return children.reduce((string, callNodeIndex) => {
+    const { name, totalTime, selfTime } = callTree.getNode(callNodeIndex);
+    const text = `\n${whitespace}- ${name} (total: ${totalTime}, self:${selfTime})`;
+    return formatTree(
+      callTree,
+      callTree.getChildren(callNodeIndex),
+      depth + 1,
+      string + text
+    );
+  }, previousString);
+}
+
+describe('focus subtree transform', function() {
+  describe('on a call tree', function() {
+    /**
+     * Assert this transformation:
+     *
+     *                     A:3,0                              C:2,0
+     *                       |                               /      \
+     *                       v       Focus [A, B, C]        v        v
+     *                     B:3,0           -->           D:1,0     F:1,0
+     *                     /    \                         |           |
+     *                    v      v                        v           v
+     *                C:2,0     H:1,0                   E:1,1       G:1,1
+     *               /      \         \
+     *              v        v         v
+     *            D:1,0     F:1,0     I:1,1
+     *            |           |
+     *            v           v
+     *          E:1,1       G:1,1
+     */
+    const profile = getProfileForUnfilteredCallTree();
+    const { dispatch, getState } = storeWithProfile(profile);
```

Comment:
be careful: because you're not creating a new store in a beforeEach, the tests are interdependent, which can be difficult to maintain and debug in the long terme.

(note: I really like how you use destructuring to get dispatch and getState :) quite surprised that it works as expected though, regarding how `this` works)